### PR TITLE
FixedPromyvions

### PR DESCRIPTION
**Pull request by Solarsurge**
*ALL Promyvion cutscenes now properly working according to mission progress.
*Fixed player dropoff after interacting with Shattered Telepoints during 2nd Promyvion
*Map acquisition now works properly in Holla and Mea

**_I affirm:_**
- [X] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [X] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/zones/Promyvion-Mea/npcs/qm1.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-----------------------------------
+
+function onTrigger(player, npc)
+    player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
+end
+
+function onTrade(player, npc, trade)
+    if npcUtil.tradeHas(trade, 1722) and not player:hasKeyItem(tpz.ki.MAP_OF_PROMYVION_MEA) then
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_MEA)
+        player:confirmTrade()
```
Comment (ibm2431) [inline scripts/zones/Promyvion-Mea/npcs/qm1.lua]: I know you're just using what's currently in these files, but while we're here, we might as well fix a retail inaccuracy~!

These should be starting an event where the screen flashes and there's text about blinding white lights and images being burned into your mind. This is event 49 for all three promyvions.

(And then give the KI and confirm the trade in onEventFinish for those events)

File: scripts/zones/Promyvion-Mea/npcs/qm1.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    else
+        player:messageSpecial(ID.text.NOTHING_HAPPENS)
```
Comment (ibm2431) [inline scripts/zones/Promyvion-Mea/npcs/qm1.lua:19]: Does this text display when you trade a bad item? Bad trades are usually just ignored.

**Review by ibm2431 — CHANGES_REQUESTED** [review]
You got close to a no-comment approval, but were sabotaged by previous scripters! 😉

File: scripts/zones/Promyvion-Mea/npcs/qm1.lua (hunk 1 above)
Comment (Solarsurge) [reply scripts/zones/Promyvion-Mea/npcs/qm1.lua]: Fixed

File: scripts/zones/Promyvion-Mea/npcs/qm1.lua (hunk 2 above)
Comment (Solarsurge) [reply scripts/zones/Promyvion-Mea/npcs/qm1.lua:19]: "Nothing happens..." is displayed on a bad trade. To my knowledge, that's correct.

**Review by ibm2431 — APPROVED** [review]
Thanks Solar!

File: scripts/zones/Promyvion-Dem/npcs/qm1.lua [hunk 3]
```diff
@@ -24,4 +25,7 @@ function onEventUpdate(player, csid, option)
 end
 
 function onEventFinish(player, csid, option)
+    if csid == 49 then
+        npcUtil.giveKeyItem(player, tpz.ki.MAP_OF_PROMYVION_DEM)
```
Comment (TeoTwawki) [inline scripts/zones/Promyvion-Dem/npcs/qm1.lua:29]: Confirm trade should also be at event finish, not before it starts

**Pull request merged**